CONTRIBUTING.md

```diff
@@ -26,7 +26,54 @@ including the linter (which checks for coding style and warnings, see [Style gui
 These checks are ran automatically using [GitHub Actions](#github-actions). You also need at least
 one approval from a maintainer - after submitting a PR, you can request a review from the top right
 Reviewers menu on the Pull Request page.
+### Branches
+Please use the following structure for creating new development branches:
+```
+[username]/[snake_case_branch_name]
+```
+This will make branches easier to find and search for.<br>
+Examples:
+```
+razvanra2/add_authoring_policy
+luigi/dev_eng_improv
+auditore/fix_md_typo
+torvalds/android_speedups
+```
+### Merging
+When developing a new feature or working on a bug, your pull request will end up containing fixup commits (commits that change the same line of code repeatedly) or too fine-grained commits. An issue that can arise from this is that the main branch history will become poluted with unnecessary commits. To avoid it, we implement and enforce a squash policy.
```

Comment:
Nit: I try to keep markdown line lengths under 100 characters.

Reply:
I'll wait for others to comment than address this, thx

Reply:
Nevermind this, I'm trying to set up a spell checker and wrapping lines manually breaks it. I think it's probably better not to wrap them.